lib/Sema/TypeCheckPropertyWrapper.cpp

```diff
@@ -406,6 +406,27 @@ PropertyWrapperTypeInfoRequest::evaluate(
     }
   }
 
+  auto diagnoseInvalidDynamicSelf = [&]() -> bool {
+    bool invalidDynamicSelf = false;
+    if (result.projectedValueVar &&
+        result.projectedValueVar->getValueInterfaceType()->is<DynamicSelfType>()) {
```

Comment:
I think you want hasDynamicSelfType() here to catch Optional<Self>, etc

Reply:
Oh, yes, thank you! 🤦🏻‍♀️